### PR TITLE
Pass sap_system and tags to inventory enpoints

### PR DESCRIPTION
**Pull request by karelhala**
## What

Enable global filter tags and sap_system to be passed down to inventory enpoints. Use `generateFilter` from frontend components (it has been added recentrly) in order to properly style the filter for inventory to be `filter[system_profile][sap_syste]: true`. This function can later help us to calculate SAP_SIDs as well.

### Before
Screenshot from 2020-10-08 16-38-38
Screenshot from 2020-10-08 16-38-46

### After
Screenshot from 2020-10-08 16-37-03
Screenshot from 2020-10-08 16-37-37

## Additional Info
[if needed]

## Code Review
@christiemolloy @ryelo @AllenBW

## Design Review
@kybaker

### CODE REVIEW

File: src/SmartComponents/SystemInventory/SystemInventoryCard.js [hunk 1]
```diff
@@ -116,7 +138,20 @@ SystemInventoryCard.propTypes = {
     fetchInventoryTotal: PropTypes.func,
     inventoryTotalSummary: PropTypes.object,
     inventoryTotalFetchStatus: PropTypes.string,
-    intl: PropTypes.any
+    intl: PropTypes.any,
+    selectedTags: PropTypes.arrayOf(PropTypes.string),
+    workloads: PropTypes.oneOf([
```
Comment (AllenBW) [inline src/SmartComponents/SystemInventory/SystemInventoryCard.js:143]: this looks like a real specific shape, should we abstract it to a constants file so all apps can consume it?

File: src/SmartComponents/SystemInventory/SystemInventoryCard.js [hunk 2]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable camelcase */
```
Comment (AllenBW) [inline src/SmartComponents/SystemInventory/SystemInventoryCard.js:1]: we do this everywhere... any qualms with killing it globally in the app?

**Review by AllenBW — COMMENTED** [review]
Just a few comments! overall love it, yer ah 👼 for doin <3

File: src/SmartComponents/SystemInventory/SystemInventoryCard.js (hunk 2 above)
Comment (karelhala) [reply src/SmartComponents/SystemInventory/SystemInventoryCard.js:1]: Fine by me, you can disable this rule and have it to warning instead.

File: src/SmartComponents/SystemInventory/SystemInventoryCard.js (hunk 1 above)
Comment (karelhala) [reply src/SmartComponents/SystemInventory/SystemInventoryCard.js:143]: Good point!

File: src/SmartComponents/SystemInventory/SystemInventoryCard.js (hunk 2 above)
Comment (AllenBW) [reply src/SmartComponents/SystemInventory/SystemInventoryCard.js:1]: Heck yeah, i can do that in a followup, no need to add it here 🤗

File: src/SmartComponents/SystemInventory/SystemInventoryCard.js (hunk 1 above)
Comment (karelhala) [reply src/SmartComponents/SystemInventory/SystemInventoryCard.js]: I've made it into complex prop type check so we have easier way of identifying what is wrong.

**Review by AllenBW — APPROVED** [review]
🥇

**Pull request merged**